### PR TITLE
#917 optional expectations samples

### PR DESCRIPTION
**Pull request by timacosta**
______________________________________
I confirm that I have read the [Contributor Agreements v1.0](https://github.com/robstoll/atrium/blob/main/.github/Contributor%20Agreements%20v1.0.txt), agree to be bound on them and confirm that my contribution is compliant.

### CODE REVIEW

**Comment by robstoll** [comment]
@timacosta let me know in case you need help

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/optionalExpectations.kt [hunk 1]
```diff
@@ -45,6 +49,8 @@ fun <E, T : Optional<E>> Expect<T>.toBePresent(): Expect<E> =
  *
  * @return an [Expect] for the subject of `this` expectation.
  *
+ * @sample ch.tutteli.atrium.api.fluent.en_GB.samples.OptionalExpectationsSamples.toBePresentFeature
```
[the remark this answers is not in the document]
Comment (timacosta) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/optionalExpectations.kt]: fixed in 
157b12f

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/optionalExpectations.kt [hunk 2]
```diff
@@ -34,6 +36,8 @@ fun <T : Optional<*>> Expect<T>.toBeEmpty(): Expect<T> =
  *
  * @return The newly created [Expect] for the inner type [E].
  *
+ * @sample ch.tutteli.atrium.api.fluent.en_GB.samples.OptionalExpectationsSamples.toBePresent
```
[the remark this answers is not in the document]
Comment (timacosta) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/main/kotlin/ch/tutteli/atrium/api/fluent/en_GB/optionalExpectations.kt]: fixed in 
157b12f

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/OptionalExpectationsSamples.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    @Test
+    fun toBeEmpty() {
+        val empty: String? = null
+        val opt = Optional.ofNullable(empty)
+
+        expect(opt).toBeEmpty()
+    }
+
+    @Test
+    fun toBePresent() {
```
[the remark this answers is not in the document]
Comment (timacosta) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/OptionalExpectationsSamples.kt]: fixed in 
157b12f

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/OptionalExpectationsSamples.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        val notEmpty: String? = "toBePresent"
+        val opt = Optional.ofNullable(notEmpty)
+
+        expect(opt).toBePresent()
+    }
+
+    @Test
+    fun toBePresentFeature() {
```
[the remark this answers is not in the document]
Comment (timacosta) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/OptionalExpectationsSamples.kt]: fixed in 
157b12f

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/OptionalExpectationsSamples.kt [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        val optX = Optional.ofNullable(x)
+        val optY = Optional.ofNullable(y)
+
+        expect(optX) {
+
+            val sum = optX.get() + optY.get()
+
+            feature("toBeLessThan") { sum }.toBeLessThan(20)
+            feature("toBeGreaterThan") { sum }.toBeGreaterThan(5)
+            feature("toBeGreaterThanOrEqualTo") { sum }.toBeGreaterThanOrEqualTo(15)
```
[the remark this answers is not in the document]
Comment (timacosta) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/OptionalExpectationsSamples.kt]: I've re-done this based on the example  AnyExpectationSamples.kt => notToEqualNull.

File: apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/OptionalExpectationsSamples.kt [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+        expect(opt).toBeEmpty()
+    }
+
+    @Test
+    fun toBePresent() {
+        val notEmpty: String? = "toBePresent"
+        val opt = Optional.ofNullable(notEmpty)
+
+        expect(opt).toBePresent()
```
[the remark this answers is not in the document]
Comment (timacosta) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/OptionalExpectationsSamples.kt]: I don't quite get the subject part, is it just in order to document how Expect<> has changed the type? At least that's what I get from the notToEqualNullFeature example. Thx

[the remark this answers is not in the document]
Comment (robstoll) [reply apis/fluent-en_GB/atrium-api-fluent-en_GB-jvm/src/test/kotlin/ch/tutteli/atrium/api/fluent/en_GB/samples/OptionalExpectationsSamples.kt]: you got it right

**Comment by robstoll** [comment]
@timacosta thanks for your contribution :+1:

**Pull request merged**